### PR TITLE
chore: updated gitconfig documentation to use the new command ensemble

### LINKED ISSUE #305: Update `gitconfig` documentation
### Is your feature request related to a problem?

We have to update the `gitconfig` documentation to use the new added `epinio gitconfig` commands.

Now also a normale user without cluster access can provide its own configuration, so no need to use directly the Kubernetes APIs creating manually secrets.

- https://docs.epinio.io/next/howtos/customization/create_git_configuration
- https://docs.epinio.io/next/references/git_configuration
- ref from testing https://github.com/epinio/epinio/pull/2521#issuecomment-1705376976

### Solution you'd like

_No response_

### Alternatives you've considered

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by andreas-kupries**
fix #305 

rewrite the gitconfig docs, use the gitconfig command instead of direct access to the cluster

### CODE REVIEW

File: docs/references/git_configuration.md [hunk 1]
```diff
@@ -5,33 +5,89 @@ title: ""
...
+It is useful to set it to the username used to generate the token.
+:::
 
-When importing from a git repository Epinio will look for the the most specific matching configuration, if any. For example when trying to clone `https://github.com/myusername/myrepo` Epinio will first look for a configuration having the `https://github.com` *URL*, a `myusername` *userOrg* and a `myrepo` *repo*.
-If not found it will look for a configuration having the `https://github.com` *URL* and a `myusername` *userOrg*. And finally it will look for a configuration having just the `https://github.com` *URL*. If no configuration is found the clone from the Git repository will run without any customization.
+## Detailed specification
 
+:::note
+Regular users have no need to know the information in this section.
+It is useful to operators though, helping in debugging and inspection.
```
Comment (jhkrug) [inline docs/references/git_configuration.md:48]:
```suggestion
This section contains information useful to operators for debugging and inspection. 
Regular users should not normally need to consult this section.
```

**Review by jhkrug — COMMENTED** [review]
Very minor suggestions.

File: docs/references/git_configuration.md [hunk 2]
```diff
@@ -5,33 +5,89 @@ title: ""
...
-- **username:** used during the Basic Authentication 
-- **password:** used during the Basic Authentication
-- **userOrg:** used to restrict the configuration to a specific organization/project 
-- **repo:** used to restrict the configuration to a specific repository 
-- **skipSSL:** used to skip the SSL verification 
-- **certificate:** the CA bundle to load for the SSL verification with self-signed certificates
+When using a PAT it has to be set as the password, and the user can be set to anything except empty.
 
+:::tip
+It is useful to set it to the username used to generate the token.
```
[the remark this answers is not in the document]
Comment (andreas-kupries) [reply docs/references/git_configuration.md]: Applied suggestion, will add a reason.

**Review by jhkrug — APPROVED** [review]
Sorry, could have approved rather than Commented last time.

**Pull request merged**